### PR TITLE
Adds new activity to cargo: Exploring The Backrooms

### PR DESCRIPTION
**Pull request by cowbot92**
# Document the changes in your pull request

Creates a new activity for cargo crew that is dedicated to exploring our randomly generated maint system colloquially known as "the backrooms".

Adds new relics for these explorers to find, and die for.

Adds a slew of hostile mobs to several tile sets in the backrooms

Maps a new area for players to enter the backrooms from, so anyone can enter! Be aware there are restrictions on what you can bring though... the portal will only allow you to bring basic clothes and a backpack, so be sure to store your gear beforehand in the lockers provided!

# Why is this good for the game?

We've had this system for a long time, and players enjoy going *to* it, and experiencing an ENTIRE Z level full of only randomly generated maint. A way to explore it each round is great.

# Testing

Will update once everything is sorted.

# Spriting

Will update once everything is sorted

# Wiki Documentation

Wiki call me

# Changelog

:cl:  
rscadd: Adds new station side activity: Backrooms exploration
rscadd: Adds new relics to the backrooms
rscadd: Adds new monsters to the backrooms
wip: This is work in progress  
imageadd: new sprites may be added
mapping: Adds a new area to enter the backrooms on station
experimental: This is experimental
/:cl:

### CODE REVIEW

**Comment by Addust** [comment]
reality hopper?

maybe make it so they can hop into different parallel realities than the backrooms

**Comment by cowbot92** [comment]
> reality hopper?
> 
> maybe make it so they can hop into different parallel realities than the backrooms

I'll add it as an alt title, but I don't know if we'll be doing more than just the backrooms at the moment for this job.

**Comment by ktlwjec0** [comment]
Just a thought, I think adding curator back to how it was and having a general explorer role for space/backrooms/etc would be cool.

**Comment by cowbot92** [comment]
> Just a thought, I think adding curator back to how it was and having a general explorer role for space/backrooms/etc would be cool.

I will consider giving curator access

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 1]
```diff
@@ -261,3 +264,72 @@
...
+		/obj/machinery/griddle = 1,
+		/obj/machinery/microwave = 1,
+		/obj/structure/closet/secure_closet/freezer/fridge = 1,
+		/obj/item/storage/box/donkpockets = 1,
+		/obj/item/kitchen/knife = 1,
+		/obj/effect/spawner/lootdrop/random_meat = 5
+
+		)
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/retaliate/goat/ = 2
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm]:
```suggestion
		/mob/living/simple_animal/hostile/retaliate/goat = 2
```

**Comment by neeshacark** [comment]
wow..... the map so good... i wonder who did it...

(i cant map review its my map)

**Comment by cowbot92** [comment]
> wow..... the map so good... i wonder who did it...
> 
> (i cant map review its my map)

cark did you add it to every map

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 2]
```diff
@@ -261,3 +264,213 @@
...
+	if(prob(75))
+		weighted_feature_spawn_list |= pick(typesof(/obj/item/circuitboard))
+	if(prob(75))
+		weighted_feature_spawn_list |= pick(typesof(/obj/item/circuitboard))
+	if(prob(75))
+		weighted_feature_spawn_list |= pick(typesof(/obj/item/circuitboard))
+	if(prob(75))
+		weighted_feature_spawn_list |= pick(typesof(/obj/item/circuitboard))
+	if(prob(75))
+		weighted_feature_spawn_list |= pick(typesof(/obj/item/circuitboard))
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:296]: use a for loop

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 3]
```diff
@@ -261,3 +264,213 @@
...
+		/mob/living/simple_animal/hostile/robot/advanced/ranged = 1
+		)
+
+/datum/dungeon_room_theme/maintenance/lasgun/pre_initialize()
+	. = ..()
+	for(var/i in 1 to 5)
+		if(prob(10))
+			weighted_feature_spawn_list[/obj/item/gun/energy/laser/captain/scattershot ]++
+		else
+			weighted_feature_spawn_list[/obj/item/melee/spear/bonespear/chitinspear]++
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:386]: just remove this entirely if it's just gonna spawn the spear like the ancient armory

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 4]
```diff
@@ -785,3 +785,11 @@
 	loot = list(/mob/living/simple_animal/hostile/asteroid/marrowweaver = 35,
 			/mob/living/simple_animal/hostile/asteroid/marrowweaver/ice = 5,
 			"" = 60)
+
+/obj/effect/spawner/lootdrop/random_anomaly_core
+	name = "anomaly core spawner"
+
+/obj/effect/spawner/lootdrop/random_anomaly_core/Initialize(mapload)
+	var/item = pick(typesof(/obj/item/assembly/signaler/anomaly))
```
Comment (Moltijoe) [inline code/game/objects/effects/spawners/lootdrop.dm:792]: use subtypesof

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 2 above)
Comment (Moltijoe) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:296]: also use subtypes of
though there's also /obj/item/circuitboard/machine, computer, and mecha that are all useless

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 5]
```diff
@@ -261,3 +264,213 @@
...
+		)
+	
+	weighted_feature_spawn_list = list(
+		/obj/structure/filingcabinet/chestdrawer = 1,
+		/obj/item/kirbyplants/random  = 2,
+		list(/obj/structure/rack , /obj/item/melee/spear/plugged_musket) = 1,
+		list(/obj/structure/rack , /obj/item/clothing/suit/armor/vest, /obj/item/clothing/head/helmet/riot) = 1,
+		/obj/structure/frame/machine = 1,
+		/obj/structure/frame/computer = 1,
+		/obj/effect/mine/stun
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:343]: don't forget the weight
i'm pretty sure if it's not included it defaults to 1
but consistency sake

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 6]
```diff
@@ -261,3 +264,213 @@
...
+		else
+			weighted_feature_spawn_list[/obj/item/melee/spear/bonespear/chitinspear]++
+
+///these don't actually spawn inside the suit storage and I think thats funny
+/datum/dungeon_room_theme/maintenance/hardsuit
+	weighted_feature_spawn_list = list(
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/mining),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/ancient),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/engine),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/medical)
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:364]: weights

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 7]
```diff
@@ -261,3 +264,213 @@
...
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/skeleton/eskimo = 2,
+		/mob/living/simple_animal/hostile/skeleton/eskimo = 1
+		)
+///I really really really want to include a real chasm but god knows im not strong enough
+/datum/dungeon_room_theme/maintenance/chasm_fake
+	weighted_possible_floor_types = list(
+		/turf/open/floor/fakepit = 3,
+		/turf/open/floor/fakepit = 5,
+		/turf/open/floor/fakepit= 3,
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:409]: what, why, why is this here 3 times with different weights

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 8]
```diff
@@ -261,3 +264,213 @@
...
+		/obj/item/stack/sheet/mineral/uranium = 5,
+		/obj/item/stack/sheet/mineral/plasma = 5,
+		/obj/item/stack/sheet/mineral/gold = 5,
+		/obj/item/stack/sheet/mineral/silver = 5,
+		/obj/item/stack/sheet/mineral/mythril = 1
+		)
+	
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/asteroid/goliath/beast = 1,
+		/mob/living/simple_animal/hostile/asteroid/goliath/beast = 1
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:431]: duplicates are pointless i'm pretty sure

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 9]
```diff
@@ -261,3 +264,213 @@
...
+/datum/dungeon_room_theme/maintenance/sus_room
+	weighted_feature_spawn_list = list(
+		/obj/item/clothing/gloves/combat = 1,
+		/obj/item/kitchen/knife/combat = 1,
+		/obj/machinery/atmospherics/components/unary/vent_pump/on = 1
+		)
+	
+	
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/retaliate/goat/suspicious = 1,
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:443]: trailing comma

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 10]
```diff
@@ -261,3 +264,213 @@
...
+		)
+///pizza tower, or spiderman 2 reference, depending on the year you time travelled from
+/datum/dungeon_room_theme/maintenance/pizza_time
+	weighted_feature_spawn_list = list(
+		/obj/item/reagent_containers/food/snacks/pizza = 1,
+		/obj/item/reagent_containers/food/snacks/pizza = 1,
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:449]: duplicates

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 11]
```diff
@@ -261,3 +264,213 @@
...
+		/obj/item/circuitboard/machine/griddle = 1,
+		/obj/item/clothing/suit/toggle/chef = 1,
+		/obj/item/clothing/suit/apron/chef = 1
+		)
+///cuackles played this once
+/datum/dungeon_room_theme/maintenance/oxygen_included
+	weighted_feature_spawn_list = list(
+		/obj/structure/tank_dispenser  = 1,
+		/obj/structure/tank_dispenser  = 1,
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:458]: duplicates

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 12]
```diff
@@ -261,3 +264,213 @@
...
+		/obj/machinery/autolathe = 1,
+		/obj/item/stack/sheet/glass/fifty = 1,
+		/obj/item/stack/sheet/metal/fifty = 1,
+		/obj/item/stack/sheet/mineral/silver/fifty = 1
+		)
+///eskimo enemy variety, they came here for the winter
+/datum/dungeon_room_theme/maintenance/eskimo
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/skeleton/eskimo = 2,
+		/mob/living/simple_animal/hostile/skeleton/eskimo = 1
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:402]: duplicates

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 13]
```diff
@@ -261,3 +264,213 @@
...
+			weighted_feature_spawn_list[/obj/item/gun/energy/laser/captain/scattershot ]++
+		else
+			weighted_feature_spawn_list[/obj/item/melee/spear/bonespear/chitinspear]++
+
+///lathes, nough said
+/datum/dungeon_room_theme/maintenance/autolathe
+	weighted_feature_spawn_list = list(
+		/obj/machinery/autolathe/hacked = 1,
+		/obj/machinery/autolathe = 1,
+		/obj/machinery/autolathe = 1,
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:393]: duplicates

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 14]
```diff
@@ -261,3 +264,213 @@
...
+	weighted_feature_spawn_list = list(
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/mining),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/ancient),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/engine),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/medical)
+		)
+
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/robot/burst = 2,
+		/mob/living/simple_animal/hostile/robot/burst = 2
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:369]: duplicates

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 15]
```diff
@@ -261,3 +264,213 @@
...
+		/obj/item/kirbyplants/random  = 2,
+		list(/obj/structure/rack , /obj/item/melee/spear/plugged_musket) = 1,
+		list(/obj/structure/rack , /obj/item/clothing/suit/armor/vest, /obj/item/clothing/head/helmet/riot) = 1,
+		/obj/structure/frame/machine = 1,
+		/obj/structure/frame/computer = 1,
+		/obj/effect/mine/stun	
+		)
+	
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/robot/burst = 2,
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:347]: trailing comma

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 16]
```diff
@@ -261,3 +264,213 @@
...
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/retaliate/goat = 2
+		)
+
+///just lik the founding fathers intended
+/datum/dungeon_room_theme/maintenance/ancient_armory
+	weighted_possible_floor_types = list(
+		/turf/open/floor/plasteel/dark = 3,
+		/turf/open/floor/plasteel = 5,
+		/turf/open/floor/plating = 3,
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:333]: trailing comma

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 17]
```diff
@@ -261,3 +264,213 @@
...
+		/obj/item/stack/cable_coil/random = 5,
+		/obj/structure/frame/machine = 3,
+		/obj/structure/frame/computer = 2,
+		/obj/effect/spawner/lootdrop/random_anomaly_core = 1,
+		/obj/effect/mine/stun
+		)
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/hivebot = 5, 
+		/mob/living/simple_animal/hostile/hivebot/range = 2,
+		/mob/living/simple_animal/hostile/hivebot/rapid = 1,
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:282]: trailing comma

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm [hunk 18]
```diff
@@ -261,3 +264,213 @@
...
+///domo arigato
+/datum/dungeon_room_theme/maintenance/machine_parts
+	weighted_feature_spawn_list = list(
+		/obj/effect/decal/cleanable/robot_debris = 2, 
+		/obj/effect/decal/cleanable/oil = 3,
+		/obj/item/stack/cable_coil/random = 5,
+		/obj/structure/frame/machine = 3,
+		/obj/structure/frame/computer = 2,
+		/obj/effect/spawner/lootdrop/random_anomaly_core = 1,
+		/obj/effect/mine/stun
```
Comment (Moltijoe) [inline code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:277]: don't forget the weight

**Review by Moltijoe — DISMISSED** [review]

**Review by AMyriad — CHANGES_REQUESTED** [review]
image

"Conflicted" with the disposals PR, pls fix

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 2 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:296]: done

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 3 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:386]: bye lasgun

File: code/game/objects/effects/spawners/lootdrop.dm (hunk 4 above)
Comment (cowbot92) [reply code/game/objects/effects/spawners/lootdrop.dm:792]: done

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 5 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:343]: done

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 6 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:364]: done

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 14 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:369]: done

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 7 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:409]: im a silly little guy

Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:409]: resolved

Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:409]: (I changed it)

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 8 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:431]: removed dup

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 9 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:443]: a bit sus but removed

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 10 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:449]: removed

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 11 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:458]: removed

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 12 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:402]: removed

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 13 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:393]: removed

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 15 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:347]: removed

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 16 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:333]: removed

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 18 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:277]: added

File: code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm (hunk 17 above)
Comment (cowbot92) [reply code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm:282]: removed

**Comment by cowbot92** [comment]
> image
> 
> "Conflicted" with the disposals PR, pls fix

I dont really know how to fix this, dont get conflicts for the map in github/vs code

**Comment by AMyriad** [comment]
> I dont really know how to fix this, dont get conflicts for the map in github/vs code

iirc you the only way to fix is redoing your changes all over again for that one map
it's a frustratingly common issue :p

**Comment by cowbot92** [comment]
> > I dont really know how to fix this, dont get conflicts for the map in github/vs code
> 
> iirc you the only way to fix is redoing your changes all over again for that one map it's a frustratingly common issue :p

I hate mapping!!!

**Review by AMyriad — APPROVED** [review]
map good

**Comment by cowbot92** [comment]
@ToasterBiome

**Review by ToasterBiome — APPROVED** [review]
